utils/s3/client.cc

```diff
@@ -97,6 +110,9 @@ client::client(std::string host, endpoint_config_ptr cfg, semaphore& mem, global
         .add_credentials_provider(std::make_unique<aws::environment_aws_credentials_provider>())
         .add_credentials_provider(std::make_unique<aws::sts_assume_role_credentials_provider>(_cfg->region, _cfg->role_arn.value_or("")))
         .add_credentials_provider(std::make_unique<aws::instance_profile_credentials_provider>());
+
+    _creds_update_timer.arm(5h);
+    _creds_invalidation_timer.arm(6h);
```

Comment:
Why hard-coded constants? Shouldn't it be `now() + _credentials.expires_at` or alike?

Reply:
Maybe I'm missing something, but at this moment `_credentials` are empty, right? So the value in the `_credentials.expires_at` is `std::chrono::steady_clock::time_point::min()`, setting any other value there will be very  confusing since the `operator bool() const` will return `false` but the `now() > _credentials.expires_at` could return `false`